src/main/java/org/spongepowered/api/item/inventory/InventoryProperty.java

```diff
@@ -25,13 +25,42 @@
 package org.spongepowered.api.item.inventory;
 
 import org.spongepowered.api.data.Property;
-import org.spongepowered.api.item.inventory.property.AbstractInventoryProperty;
+import org.spongepowered.api.util.ResettableBuilder;
 
 /**
- * Base interface for Inventory Properties. It is recommended that implementors
- * subclass {@link AbstractInventoryProperty} instead of implementing this
- * interface directly. 
+ * Base interface for Inventory Properties.
  */
 public interface InventoryProperty<K, V> extends Property<K, V> {
 
+    /**
+     * Represents a builder class to create Inventory Properties
+     *
+     * @param <T>
+     * @param <B>
+     */
+    interface Builder<V, T extends InventoryProperty<?, V>, B extends Builder<V, T, B>> extends ResettableBuilder<T, B> {
+
```

Comment:
A builder method for the key?

Reply:
I'm considering removing that.
It is always the default key atm.

Reply:
Yeah, maybe remove the key from the API, haven't seen it used anywhere that it isn't default.
@gabizou ?